### PR TITLE
refactor: use block.timestamp instead of block.number

### PR DESCRIPTION
**Pull request by xorsal**
# 🤖 Linear

We probably need to update events as well.

Closes OPO-647

### CODE REVIEW

**Comment by xorsal** [comment]
Tried bumping husky to fix a weird error I'm getting in `prophet-modules` when I use the dependency`@defi-wonderland/prophet-core#de18a05` but husky v9 dropped support Node 14 and 16.

File: package.json
Comment (gas1cent) [inline package.json:1]: Could you cherry-pick the husky-related changes to a separate PR? This way we will be able to revert them if they cause any trouble down the road (as we have experienced with Husky already).

Comment (gas1cent) [reply package.json:1]: We should update all github actions as well to use node v20. As you've pointed out above, versions 14 and 16 are obsolete.

Comment (gas1cent) [reply package.json:1]: Also something to take into account: https://github.com/defi-wonderland/solidity-foundry-boilerplate/pull/84/files#r1778314008

Comment (xorsal) [reply package.json:1]: Yes. I was trying to fix the error I mentioned in the comments. My original idea was to rebase this branch, put the husky bump commit first, and then drop it from this branch.

File: solidity/contracts/Oracle.sol [hunk 1]
```diff
@@ -132,7 +132,7 @@ contract Oracle is IOracle {
     _participants[_requestId] = abi.encodePacked(_participants[_requestId], _response.proposer);
     IResponseModule(_request.responseModule).propose(_request, _response, msg.sender);
     _responseIds[_requestId] = abi.encodePacked(_responseIds[_requestId], _responseId);
-    responseCreatedAt[_responseId] = uint128(block.number);
+    responseCreatedAt[_responseId] = uint128(block.timestamp);
 
     emit ResponseProposed(_requestId, _responseId, _response, block.number);
```
Comment (0xJabberwock) [inline solidity/contracts/Oracle.sol:137]: It should be questioned whether emitting the `block.number` or `block.timestamp` is useful anymore.

Comment (xorsal) [reply solidity/contracts/Oracle.sol:137]: >It should be questioned whether emitting the block.number or block.timestamp is useful anymore.

I agree. Initially I thought block.number was unnecessary and changing it to emit block.timestamp was better, but in truth, both values can be easily obtained. I wonder if it changes anything for the offchain team.

>Also, does it make sense to cast to uint128?

This one seems altogether unnecessary. I couldn't find a specific use for it in any of the four mappings using uint128 block.numbers (now timestamps as per this PR).

**Review by gas1cent — APPROVED** [review]
Beauty 🙌🏻

**Review by 0xJabberwock — APPROVED** [review]

**Pull request merged**